pstress/generate_rocksdb_options.sh

```diff
@@ -0,0 +1,176 @@
+
+# Create an output file which contains all the options/values
+rm -rf $OUTPUT_FILE
+touch $OUTPUT_FILE
+
+while read line; do
+  OPTION="--loose-$(echo $line | awk '{print $1}')"
+  VALUE="$(echo $line | awk '{print $2}')"
+  if [ "$VALUE" == "" ]; then
+    echoit "Working on option '$OPTION' which has no default value..."
+  else
+    echoit "Working on option '$OPTION' with default value '$VALUE'..."
+  fi
+  if [[ " ${EXCLUDED_LIST[@]} " =~ " ${OPTION} " ]]; then
+    echoit "Option '$OPTION' is logically excluded from being handled by this script..."
+  elif [[ "$OPTION" == "--loose-rocksdb_access_hint_on_compaction_start" ]]; then
+    echoit " > Adding possible values NONE, NORMAL, SEQUENTIAL, WILLNEED for option '${OPTION}' to the final list..."
+    echo "$OPTION=NONE" >> $OUTPUT_FILE
+    echo "$OPTION=NORMAL" >> $OUTPUT_FILE
+    echo "$OPTION=SEQUENTIAL" >> $OUTPUT_FILE
+    echo "$OPTION=WILLNEED" >> $OUTPUT_FILE
+  elif [[  "$OPTION" == "--loose-rocksdb_index_type" ]]; then
+    echoit " > Adding possible values kBinarySearch, kHashSearch for option '${OPTION}' to the final list..."
+    echo "$OPTION=kBinarySearch" >> $OUTPUT_FILE
+    echo "$OPTION=kHashSearch" >> $OUTPUT_FILE
+  elif [[ "$OPTION" == "--loose-rocksdb_write_policy" ]]; then
+    echoit " > Adding possible values write_committed, write_prepared, write_unprepared for option '${OPTION}' to the final list..."
+    echo "$OPTION=write_committed" >> $OUTPUT_FILE
+    echo "$OPTION=write_prepared" >> $OUTPUT_FILE
+    echo "$OPTION=write_unprepared" >> $OUTPUT_FILE
+  elif [[ "$OPTION" == "--loose-rocksdb_manual_compaction_bottommost_level" ]]; then
+    echoit " > Adding possible values kSkip, kIfHaveCompactionFilter, kForce, kForceOptimized for option '${OPTION}' to the final list..."
+    echo "$OPTION=kSkip" >> $OUTPUT_FILE
+    echo "$OPTION=kIfHaveCompactionFilter" >> $OUTPUT_FILE
+    echo "$OPTION=kForce" >> $OUTPUT_FILE
+    echo "$OPTION=kForceOptimized" >> $OUTPUT_FILE
+  elif [[ "$OPTION" == "--loose-rocksdb_cache_high_pri_pool_ratio" ]]; then
+    echoit " > Adding possible values 0, 0.25, 0.5, 0.75, 1.0 for option '${OPTION}' to the final list..."
+    echo "$OPTION=0.0" >> $OUTPUT_FILE
+    echo "$OPTION=0.25" >> $OUTPUT_FILE
+    echo "$OPTION=0.50" >> $OUTPUT_FILE
+    echo "$OPTION=0.75" >> $OUTPUT_FILE
+    echo "$OPTION=1.0" >> $OUTPUT_FILE
+  elif [[ "$OPTION" == "--loose-rocksdb_info_log_level" ]]; then
+    echoit " > Adding possible values (debug_level, info_level, warn_level, error_level, fatal_level ) for option '${OPTION}' to the final list..."
+    echo "$OPTION=debug_level" >> $OUTPUT_FILE
+    echo "$OPTION=info_level" >> $OUTPUT_FILE
+    echo "$OPTION=warn_level" >> $OUTPUT_FILE
+    echo "$OPTION=error_level" >> $OUTPUT_FILE
+    echo "$OPTION=fatal_level" >> $OUTPUT_FILE
```

Comment:
this option can be skipped from randomness?

Reply:
Yeah. The more info the error log has, the better it is.
Thanks for the suggestion.